tests/test_io.py

```diff
@@ -337,3 +342,70 @@ def test_feather():
     check(assert_type(DF.to_feather(bio), None), type(None))
     bio.seek(0)
     check(assert_type(read_feather(bio), DataFrame), DataFrame)
+
+
+def test_read_sql():
+    with ensure_clean() as path:
+        con = sqlite3.connect(path)
+        check(assert_type(DF.to_sql("test", con=con), Union[int, None]), int)
+        check(
+            assert_type(read_sql("select * from test", con=con), DataFrame), DataFrame
+        )
+        con.close()
+
+
+def test_read_sql_generator():
+    with ensure_clean() as path:
+        con = sqlite3.connect(path)
+        check(assert_type(DF.to_sql("test", con=con), Union[int, None]), int)
+
+        check(
+            assert_type(
+                read_sql("select * from test", con=con, chunksize=1),
+                Generator[DataFrame, None, None],
+            ),
+            Generator,
+        )
+        con.close()
+
+
+def test_read_sql_table():
+    if TYPE_CHECKING:
```

Comment:
It's a bit surprising that the other tests don't require `if TYPE_CHECKING` and this one does.  Do you know why?

Reply:
Because sqlite3 doesn't have read_table, so it can't be used in this test.  The other 2 work with sqlite3.

Reply:
Can you add a comment about that?

Reply:
Still would like a comment in the code as to why `TYPE_CHECKING` is used here

Reply:
Added a comment.